packages/http/test/responses.test.ts

```diff
@@ -64,6 +64,24 @@ describe("body resolution", () => {
   });
 });
 
+describe("response cookie", () => {
+  it("emit diagnostics for implicit @cookie in the response", async () => {
+    const [_, diagnostics] = await compileOperations(`
+        op get(): { @cookie token: string };
+      `);
+
+    expectDiagnostics(diagnostics, { code: "@typespec/http/response-cookie-not-supported" });
+  });
+
+  it("doesn't emit response-cookie-not-supported diagnostics for explicit @cookie in the response", async () => {
```

Comment:
@markcowl  @timotheeguerin  a little bit wired for me - because cookie should be in `set-cookie` headers in response but the `@body` decorator stands for http body.

why we allow a cookie property in `@body` not `@header`?

Reply:
Apologies for the lack of clarity in my explanation.

If we allow the `@cookie` property to be included as ignored metadata in the implicit body, it could lead to a breaking change in the future if TypeSpec begins to support `set-cookie` in responses.
For this reason, we decided to diagnose it as `response-cookie-not-supported` and exclude it from the response body.

You can find more details in the [review discussion](https://github.com/microsoft/typespec/pull/4761#discussion_r1805082132).

Reply:
NOTE: The reason I allow `@cookie` in the explicit body is that other parameters are handled in the same way.

This approach does not introduce breaking changes, and we cannot just omit only cookies while consistently including other `metadata-ignored` properties.